tests/out/hlsl/push-constants.hlsl

```diff
@@ -0,0 +1,21 @@
+
+struct PushConstants {
```

Comment:
How does this interact with vertex shaders' builtin constants? It would be useful to put that into a test.